climada/entity/exposures/base.py

```diff
@@ -1037,11 +1040,15 @@ def affected_total_value(self, hazard):
             a centroids is assigned
 
         """
+        assigned_centroids = self.gdf[hazard.centr_exp_col]
         nz_mask = (
             (self.gdf.value.values > 0)
-            & (self.gdf[hazard.centr_exp_col].values >= 0)
+            & (assigned_centroids.values >= 0)
         )
-        return np.sum(self.gdf.value.values[nz_mask])
+        cents = np.unique(assigned_centroids[nz_mask])
+        cent_with_inten_above_thres = (hazard.intensity[:,cents] > threshold_affected).sum(axis=0).nonzero()[1]
```

Comment:
Since it's not important which event surpassed the threshold, it might make sense to compute the row-wise maximum first, and then query `nonzero`:
```suggestion
        cent_with_inten_above_thres = (hazard.intensity[:,cents].max(axis=0) > threshold_affected).nonzero()[1]
```
(I did not benchmark this)

Reply:
Good idea! I will benchmark it.

Reply:
I meant *column-wise*, of course (i.e., the maximum intensity at every centroid), but `axis=0` should mean exactly that 🙇

Reply:
My benchmarking shows a slight improvement, so we keep the new version. Thanks!